### ISSUE TITLE
Kane's ITM.GRADE field is printed twice on invoices, one in brackets and the other without brackets.

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1698]
_Imported from trac ticket 1698.
Created by **wayne**
Opened in trac: Jun 5, 2012, 10:06:29 AM
Last modified in trac: Jul 23, 2012, 4:13:03 PM_
***
_by wayne:_
Kent,
Good morning, the ITM.GRADE field is printed twice on our invoices, one in brackets and the other without brackets. Can you remove the one with brackets? See attachment.
Thanks,
*Wayne Liburd*

[attachment:"0531208WR16.pdf"]

***
* kbower attached [0531208WR16.pdf](https://trac.retailarchitects.com/trac/attachment/ticket/1698/0531208WR16.pdf) on Jun 5, 2012, 10:06:29 AM
* kbower attached [None](https://trac.retailarchitects.com/trac/attachment/ticket/1698/None) on Jul 23, 2012, 3:33:29 PM

**Comment by kbower** [comment]
_**criley** trac comment on Jul 18, 2012, 1:07:09 PM:_

(Changeset [[2885]](../commit/8d89538ac6baa835df388ae7f70a7f9e36e29249))
*    Fixes #1698  (Kane's ITM.GRADE field is printed twice on invoices, 
          one in brackets and the other without brackets.)

**Comment by kbower** [comment]
_**criley** trac status change on Jul 18, 2012, 1:07:09 PM:_ **closed**

**Comment by kbower** [comment]
_**criley** trac status change on Jul 18, 2012, 1:26:37 PM:_ **reopened**

**Comment by kbower** [comment]
_**criley** trac comment on Jul 18, 2012, 1:53:14 PM:_

(Changeset [[2886]](../commit/c35819b146d62e7cd4472b0e0f79517a9b9db775))
*    Fixes #1698 (Small tweak)

**Comment by kbower** [comment]
_**criley** trac status change on Jul 18, 2012, 1:53:14 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 3:31:47 PM:_

Crash:
```
  File "/home/rarch/trunk/src/appserver/pylotengine/templates/../../reports/kan_receipt.html", line 248, in <Expression '?'>
    <py:otherwise py:with="desc = line.product.displaydescription.replace('(' + line.product.grade + ')', line.product.grade)">
TypeError: cannot concatenate 'str' and 'NoneType' objects
```

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 23, 2012, 3:31:47 PM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 3:33:29 PM:_


[[Image(None)]]

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 4:03:06 PM:_

Replying to [kbower](comment:7):
> Crash:
> {{{
>   File "/home/rarch/trunk/src/appserver/pylotengine/templates/../../reports/kan_receipt.html", line 248, in <Expression '?'>
>     <py:otherwise py:with="desc = line.product.displaydescription.replace('(' + line.product.grade + ')', line.product.grade)">
> TypeError: cannot concatenate 'str' and 'NoneType' objects
> }}}


This was fixed in [r2906](../commit/c9c1aec5aee9d8e0fa6363febf9696cbb1afe0d1)

**Comment by kbower** [comment]
_**kbower** trac comment on Jul 23, 2012, 4:13:03 PM:_

(Changeset [[2909]](../commit/a6cda2252711ebb6f66687ddc6eaf98631323b61))
 * fixes #1698 Kane's print grade only once (just needed product.description instead of product.displaydescription)

**Comment by kbower** [comment]
_**kbower** trac status change on Jul 23, 2012, 4:13:03 PM:_ **closed**
